perun-core/src/main/java/cz/metacentrum/perun/taskslib/dao/jdbc/TaskResultDaoJdbc.java

```diff
@@ -287,26 +287,61 @@ public int clearOld(int engineID, int numDays) {
 	private static byte[] clearZeroBytesFromString(byte[] data, int maxLength) {
 		if (data == null) return null;
 		ByteArrayOutputStream dataOut = new ByteArrayOutputStream() ;
-		int limit = (maxLength < data.length) ? maxLength - 4 : data.length;
+		int limit = (maxLength < data.length) ? maxLength - MAX_NUMBER_OF_UTF8_BYTES : data.length;
 		for (int i = 0; i < limit; i++) {
 			if (data[i] != 0x00)
 				dataOut.write(data[i]);
 		}
 		if(maxLength < data.length) {
```

Comment:
This boolean value is there twice. I would add a variable with good description in name. Something like:
`boolean dataExceedMaxLength = maxLength < data.length;`
And then use this variable in both if clause.